### PR TITLE
use client config when configuring grpc client provider

### PR DESCRIPTION
**Pull request by podlech**

### CODE REVIEW

**Review by swankjesse — APPROVED** [review]

**Comment by swankjesse** [comment]
Nice find.

File: misk/src/main/kotlin/misk/client/OkHttpClientCommonConfigurator.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.google.inject.Singleton
+import okhttp3.ConnectionPool
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient.Builder
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+@Singleton
+class OkHttpClientCommonConfigurator @Inject internal constructor() {
+  fun configure(builder: Builder, config: HttpClientEndpointConfig): Builder {
```
Comment (frojasg) [inline misk/src/main/kotlin/misk/client/OkHttpClientCommonConfigurator.kt:13]: Should we inject `Provider<HttpClientsConfig>` so we don't need to pass `HttpClientEndpointConfig` as an argument?

**Review by frojasg — APPROVED** [review]
LGTM!

File: misk/src/main/kotlin/misk/client/OkHttpClientCommonConfigurator.kt (hunk 1 above)
Comment (podlech) [reply misk/src/main/kotlin/misk/client/OkHttpClientCommonConfigurator.kt:13]: Unfortunately [HttpClientFactory.get](https://github.com/podlech/misk/blob/ce105346cf587e473ef5e46c3335ad3235e2cab5/misk/src/main/kotlin/misk/client/HttpClientFactory.kt#L28) receives the config as a parameter, so I'll keep it as is.

**Review by swankjesse — APPROVED** [review]

**Pull request merged**